nucypher/utilities/clouddeploy.py

```diff
@@ -289,16 +292,50 @@ def chain_name(self):
     def inventory_path(self):
         return os.path.join(DEFAULT_CONFIG_ROOT, NODE_CONFIG_STORAGE_KEY, f'{self.namespace_network}.ansible_inventory.yml')
 
-    def generate_ansible_inventory(self, node_names, **kwargs):
+    def update_generate_inventory(self, node_names, **kwargs):
+
+        # filter out the nodes we will not be dealing with
+        nodes = {key: value for key, value in self.config['instances'].items() if key in node_names}
+        if not nodes:
+            raise KeyError(f"No hosts matched the supplied names: {node_names}.  Try `nucypher cloudworkers list-hosts`")
+
+        default_envvars = [
+            ('NUCYPHER_KEYRING_PASSWORD',  self.config['keyringpassword']),
+            ('NUCYPHER_WORKER_ETH_PASSWORD', self.config['ethpassword']),
+        ]
+
+        input_envvars = [(k, v) for k, v in self.envvars]
+
+        # populate the specified environment variables as well as the
+        # defaults that are only used in the inventory
+        for key, node in nodes.items():
+            node_vars = nodes[key].get('runtime_envvars', {})
+            for k, v in input_envvars:
+                node_vars.update({k: v})
+            nodes[key]['runtime_envvars'] = node_vars
+
+            # we want to update the config with the specified envvars
+            # so they will persist in future invocations
+            self.config['instances'][key] = copy.deepcopy(nodes[key])
+
+        # we don't want to save the default_envvars to the config file
+        # but we do want them to be specified to the inventory template
+        # but overridden on a per node basis if previously specified
```

Comment:
A staker who adds their own envars might expect all others/defaults to be in the config file, especially if they are being used for something, so this should be really spelled out in documentation

Reply:
this refers to the *worker* config file, not the NuCypher ursula.json

There are two envars that are _required_, specifically the keychain and worker passwords.
Since these are generally handled by default for all nodes in a namespace, we don't want to write them on a per node basis in the worker_config.json, since it will generally be the same thing for every node and also, the less times we write passwords into files the better imo.

But this bit of code does allow them to be overridden on a per host basis and _only then_ would they be written to the config file.